.github/workflows/npm-publish.yml

```diff
@@ -4,8 +4,12 @@
 name: Node.js Package
 
 on:
-  release:
-    types: [published]
+  push:
+    tags:
+      - '\d+.\d+.\d+'
```

Comment:
I did not implement validation of version number due to restriction of pushed tags. Alternativly, any tag can be allowed and validated with a semver regex